File: samples/python/mistralai/streaming.py
"""For a better user experience, you will want to stream the response of the model
so that the first token shows up early and you avoid waiting for long responses."""

import os
from mistralai import Mistral

token = os.environ["GITHUB_TOKEN"]
endpoint = "https://models.github.ai/inference"

# Pick one of the Mistral models from the GitHub Models service
model_name = "mistral-small-2503"

# Create a client
client = Mistral(api_key=token, server_url=endpoint)

# Call the chat completion API
response = client.chat.stream(
    model=model_name,
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {
            "role": "user",
            "content": "Give me 5 good reasons why I should exercise every day.",
        },
    ],
)

# Print the streamed response
for event in response:
    if event.data.choices:
        print(event.data.choices[0].delta.content or "", end="")

print()

File: samples/js/mistralai/streaming.js
import MistralClient from '@mistralai/mistralai';

const token = process.env["GITHUB_TOKEN"];
const endpoint = "https://models.github.ai/inference/";

/* Pick one of the Mistral models from the GitHub Models service */
const modelName = "mistral-ai/Mistral-small";

export async function main() {

  const client = new MistralClient(token, endpoint);

  const response = await client.chatStream({
    model: modelName,
    messages: [
      { role:"system", content: "You are a helpful assistant." },
      { role:"user", content: "Give me 5 good reasons why I should exercise every day." }
    ],
  });

  for await (const chunk of response) {
    if (chunk.choices[0].delta.content !== undefined) {
      const streamText = chunk.choices[0].delta.content;
      process.stdout.write(streamText);
    }
  }
}

main().catch((err) => {
  console.error("The sample encountered an error:", err);
});